### ISSUE TITLE
Failure to create storage account using --auth-mode login

### ISSUE THREAD

**Issue by mvalijipa** [microsoft/AzureTRE#3235]
When executing the ‘make all’ command we encounter an issue related to the authentication mode when trying to create a storage account (codesnippet: AzureTRE > devops > terraform > bootstrap.sh > ln16 - ln25 ).
By default, the ‘--auth-mode’ is set to ‘login’—when executing this we see the below error related to network rules.
![image](https://user-images.githubusercontent.com/107870922/220132082-a2e45095-0764-4925-a1fa-e828c9f79f35.png)
The error message indicates the use of ‘networkRuleSet’ to enable permissions. However, this also does not resolve the issue.
Only approach that allowed us to create the storage account was to change the authentication mode from --auth-mode login to --account-key.

**Comment by tamirkamara** [comment]
Have you tried to rerun the command with the same config after it first failed?
If not, can you try now and see if the error persists?

**Comment by mvalijipa** [comment]
Yes, I have tried to rerun this and just tried this again with the latest repository pull. Still receiving the same error message as mentioned above.

**Comment by tamirkamara** [comment]
Are you running this via GitHub CodeSpaces?

**Comment by mvalijipa** [comment]
No, currently using dev containers in VSCode. But I get the same issue even if I use CodeSpaces.

**Comment by marrobi** [comment]
@mvalijipa are there any corporate Azure policies applied to your subscription that block access to storage accounts? Can you look at the networking settings in the Azure portal for the storage account and report back? 

Also can you check your permissions on the storage account in IAM in the azure portal, and report back with your permissions.

**Comment by mvalijipa** [comment]
@marrobi I am currently using my own tenant and MSDN subscription to deploy. These are the network settings: 
![image](https://user-images.githubusercontent.com/107870922/220879971-e61500e4-fa84-4d94-8f0b-ac41deb4a8f6.png)

And this my access on the storage account:
![image](https://user-images.githubusercontent.com/107870922/220880360-752f26d9-649c-46ce-8411-c8e2b654be07.png)

**Comment by marrobi** [comment]
Very odd.

Can you try something like the blow from your dev container and see what you get?

```
az group create -n "tmprg" --location "westeurope"

az storage account create --resource-group "tmprg" \
--name "tmpsamr1234" --location "westeurope" \
--kind StorageV2 --sku Standard_LRS -o table

az storage container create --account-name  "tmpsamr1234" \
--name "tmpcontainer" \
--auth-mode login -o table
```

Don't see think this is an Azure TRE specific issue.

**Comment by mvalijipa** [comment]
As suggested, I ran the above commands using the CLI via Azure portal and I get a similar error when creating the storage container. 

![image](https://user-images.githubusercontent.com/107870922/220962448-57908e8f-f37f-49b6-a860-75273baa23f9.png)

**Comment by marrobi** [comment]
So this issue is not related to Azure TRE. Can you try the same three commands in cloud shell?

https://learn.microsoft.com/en-us/azure/cloud-shell/overview

**Comment by mvalijipa** [comment]
Yep same issue with Cloud Shell. I tried to update the 'networkRuleSet' but that hasnt helped as well.

Only way I get it to work is to use the `--account-key "$SA_KEY" ` rather than `--auth-mode login`

**Comment by marrobi** [comment]
Can you add yourself as "Storage Blob Data Contributor" to the storage account and then try creating the container again.

**Comment by marrobi** [comment]
Ahh, got it. For some reason your are not using ARM IAM on the subscription, you only have the classic Co-Administrator role. Is this a VERY old subscription?

![image](https://user-images.githubusercontent.com/17089773/220968844-0ab842c2-42f5-446c-85cd-3441095505d4.png)

Can you add yourself via IAM to the Owner role for the subscription.
